### ISSUE TITLE
type argument in scale_fill_continuous but not in scale_color_continuous

### ISSUE THREAD

**Issue by katossky** [tidyverse/ggplot2#3053]
`scale_fill_continuous` has a `type` argument (which can be set to `"viridis"` or `"gradient"`), but not `scale_color_continous`. I don't grasp the reason why.

Since `scale_..._continuous` is a common repeated pattern accross `ggplot2`, I find it strange to force people to move to more exotic functions such as `scale_colour_viridis_c` when they have to use colors – besides the wierdness of the asymetry between `scale_colour_continuous` and `scale_fill_continuous`.

I am not sure what the solution should be, but I'm uncomfortable with the current state. I feel like a `type` argument in `scale_color_continuous` would be a first step.

**Comment by katossky** [comment]
Now I'm confused reading the documentation again. Sorry.

**Comment by katossky** [comment]
Reproducible exemple:

    iris %>% ggplot(aes(x=Petal.Length, y=Petal.Width, fill=Sepal.Length)) + geom_point(shape=21) + scale_fill_continuous(type='viridis') #works
   iris %>% ggplot(aes(x=Petal.Length, y=Petal.Width, color=Sepal.Length)) + geom_point(shape=16) + scale_color_continuous(type='viridis') # doesn't

**Comment by clauswilke** [comment]
It's a bug, specifically in this line: https://github.com/tidyverse/ggplot2/blob/1c09bae2aa24320bcb4891c664cccc16efc86a8a/R/zxx.r#L115

`scale_color_continuous` is defined as `scale_colour_gradient`, but it should be defined as `scale_colour_continuous`. Note that things work as stated in the documentation when you use `scale_colour_continuous`.

``` r
library(ggplot2)

# works
ggplot(iris, aes(x=Petal.Length, y=Petal.Width, fill=Sepal.Length)) + 
  geom_point(shape=21) + scale_fill_continuous(type='viridis')
```

![](https://i.imgur.com/gQyzcdH.png)

``` r

# works
ggplot(iris, aes(x=Petal.Length, y=Petal.Width, color=Sepal.Length)) + 
  geom_point(shape=16) + scale_colour_continuous(type='viridis')
```

![](https://i.imgur.com/yCh7EOp.png)

``` r

# doesn't work
ggplot(iris, aes(x=Petal.Length, y=Petal.Width, color=Sepal.Length)) + 
  geom_point(shape=16) + scale_color_continuous(type='viridis')
#> Error in continuous_scale(aesthetics, "gradient", seq_gradient_pal(low, : unused argument (type = "viridis")

scale_color_continuous <- scale_colour_continuous
# now works
ggplot(iris, aes(x=Petal.Length, y=Petal.Width, color=Sepal.Length)) + 
  geom_point(shape=16) + scale_color_continuous(type='viridis')
```

![](https://i.imgur.com/AUAHRvb.png)

<sup>Created on 2019-01-02 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by hlendway** [comment]
I will work on this for dev day :-)

**Comment by clauswilke** [comment]
Closed via #3079.
